### ISSUE TITLE
"Unexpected flow node flags" probably related to overload+TypeVar

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#2775]
**Describe the bug**
The issue occurs on the [pandas CI](https://github.com/pandas-dev/pandas/runs/4661438307?check_suite_focus=true) for [this PR](https://github.com/pandas-dev/pandas/pull/45111) with [this commit](https://github.com/pandas-dev/pandas/pull/45111/commits/8fd8b1fad5a1656999b7364565638d686ae9671d) with pyright 1.1.200.

The error printed by pyright:

```
No configuration file found.
pyproject.toml file found at /home/runner/work/pandas/pandas.
Loading pyproject.toml file at /home/runner/work/pandas/pandas/pyproject.toml
Assuming Python platform Linux
Searching for source files
Found 315 source files
An internal error occurred while type checking file "/home/runner/work/pandas/pandas/pandas/io/excel/_base.py": Error: Debug Failure. Unexpected flow node flags
    at narrowConstrainedTypeVarRecursive (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1218:17)
    at Object.narrowConstrainedTypeVar (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1223:41)
    at Object.narrowConstrainedTypeVar (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:22189:31)
    at x.visitReturn (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:806:74)
    at x.visitNode (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:390:26)
    at x.walk (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:97:37)
    at x.walk (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:244:19)
    at forEach (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:106:22)
    at Array.forEach (<anonymous>)
    at x.walkMultiple (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:104:15)
Error performing analysis: Error: Debug Failure. Unexpected flow node flags
    at narrowConstrainedTypeVarRecursive (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1218:17)
    at Object.narrowConstrainedTypeVar (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/codeFlowEngine.ts:1223:41)
    at Object.narrowConstrainedTypeVar (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:22189:31)
    at x.visitReturn (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:806:74)
    at x.visitNode (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:390:26)
    at x.walk (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:97:37)
    at x.walk (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:244:19)
    at forEach (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:106:22)
    at Array.forEach (<anonymous>)
    at x.walkMultiple (/opt/hostedtoolcache/node/16.13.1/x64/lib/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:104:15)
```

I hoped that the following toy example would trigger the same issue, but it doesn't:
```py
from __future__ import annotations

from typing import TypeVar, overload

IntStrT = TypeVar("IntStrT", int, str)


@overload
def test(sheet_name: str | int = ...) -> int:
    ...


@overload
def test(sheet_name: list[IntStrT] = ...) -> dict[IntStrT, int]:
    ...


def test(
    sheet_name: str | int | list[int] | list[str] = 0,
) -> int | dict[str, int] | dict[int, int]:
    ...


reveal_type(test())
reveal_type(test(sheet_name=0))
reveal_type(test(sheet_name=[0]))
reveal_type(test(sheet_name="a"))
reveal_type(test(sheet_name=["a"]))
```

**To Reproduce**
Steps to reproduce the behavior.

```sh
git clone -b typ_excel_writer git@github.com:phofl/pandas.git
cd pandas
git checkout 8fd8b1fad5a1656999b7364565638d686ae9671d
pyright
```

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. This will be fixed in the next version of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.202, which I just published. It will also be included in the next release of pylance.
